### PR TITLE
Add Websocket to GameState component. 

### PR DESCRIPTION
**Pull request by arimet**
Implementation of websocket to display in real time our game board

### CODE REVIEW

File: src/GameState/index.tsx [hunk 1]
```diff
@@ -35,6 +34,8 @@ const reducer = (
   return { ...state, [action.type]: action.value };
 };
 
+let WS: any = undefined;
```
Comment (alexisjanvier) [inline src/GameState/index.tsx]: Why do you need to declare `WS` outside the `GameState` method?

File: src/GameState/index.tsx [hunk 2]
```diff
@@ -96,10 +97,34 @@ export default function GameState({ navigation, route, player }: any) {
...
+        title: 'Error',
+        status: 'error',
+        description: 'An error occurred with the live function',
+      });
+    };
+  };
+
   useEffect(() => {
     getGame();
+    initWebSocket();
```
Comment (alexisjanvier) [inline src/GameState/index.tsx:122]: To close your websocket, you must do it as return of your useEffect.

```javascript
useEffect(() => {
    getGame();
    initWebSocket();
    return () => {
        if (WS) WS.close();
     };
}, []);
```

https://fr.reactjs.org/docs/hooks-reference.html#cleaning-up-an-effect

**Review by alexisjanvier — CHANGES_REQUESTED** [review]

File: src/GameState/index.tsx [hunk 3]
```diff
@@ -96,8 +96,33 @@ export default function GameState({ navigation, route, player }: any) {
...
   };
 
+  const initWebSocket = () => {
+    WS = new WebSocket(GATEWAY_URL);
+    WS.onopen = () => {
+      WS.send(JSON.stringify({ event: 'initGame', data: { gameId } }));
+    };
+
+    WS.onmessage = (e: any) => {
+      const gameState = JSON.parse(e.data)?.gameState as Game;
```
Comment (alexisjanvier) [inline src/GameState/index.tsx:106]: What happens if there is no data or if the json is incorrectly formatted?

File: src/GameState/index.tsx [hunk 4]
```diff
@@ -96,8 +96,33 @@ export default function GameState({ navigation, route, player }: any) {
     navigation.goBack();
   };
 
+  const initWebSocket = () => {
```
Comment (alexisjanvier) [inline src/GameState/index.tsx:99]: What happens if the websocket server is not available?

File: src/GameState/index.tsx [hunk 5]
```diff
@@ -96,8 +96,33 @@ export default function GameState({ navigation, route, player }: any) {
...
+    WS.onopen = () => {
+      WS.send(JSON.stringify({ event: 'initGame', data: { gameId } }));
+    };
+
+    WS.onmessage = (e: any) => {
+      const gameState = JSON.parse(e.data)?.gameState as Game;
+      dispatch({ type: 'game', value: gameState });
+    };
+
+    WS.onerror = (e: any) => {
```
Comment (alexisjanvier) [inline src/GameState/index.tsx:110]: The `onerror` catches all errors, including connection initialization errors?

File: src/GameState/index.tsx (hunk 3 above)
Comment (arimet) [reply src/GameState/index.tsx:106]: I added a check in the following PR

File: src/GameState/index.tsx (hunk 4 above)
Comment (arimet) [reply src/GameState/index.tsx:99]: You're right, right now I don't have anything to wrestle with. 
In the following PR I added something like: `ws.addEventListener('error', (err) => console.log(err.message));`

**Review by alexisjanvier — APPROVED** [review]
Ok, let's see in the next PR

**Pull request merged**